include/net/nrf_cloud_agps.h

```diff
@@ -63,6 +63,18 @@ void nrf_cloud_agps_processed(struct nrf_modem_gnss_agps_data_frame *received_el
  */
 bool nrf_cloud_agps_request_in_progress(void);
 
+/**
+ * @brief Set use case and mask angle for filtered ephemerides mode.
+ *
+ * @param use_case Base use case to set in GNSS unit; function will attempt to
+ *        set this as well as NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE, but
+ *        fall back to input value if the modem firmware present does not
+ *        support it.
+ * @param mask_angle Elevation threshold angle to set in GNSS unit.
+ *
+ * @return 0 if successful, otherwise a (negative) error code.
+ */
+int nrf_cloud_agps_configure_filter(uint8_t use_case, uint8_t mask_angle);
```

Comment:
I understand that the same code is used in multiple places and it would be nice to have it in one place, but I don't think this belongs in the nRF Cloud library. I find it odd that a nRF Cloud library function is used to configure GNSS. Also, the actual functionality may not be what we really want.

It makes sense to disable scheduled downloads when A-GPS is used. However, it's not really related to the use of filtered ephemerides. The main reason for disabling scheduled downloads are almanacs, not ephemerides. Actually, GNSS only does scheduled downloads for ephemerides when the used fix interval is 30 minutes or less. And as mentioned by @hiltunent in a previous comment, it does not make sense to use filtered ephemerides with this short intervals.

It's also not mandatory to set the GNSS elevation threshold angle to the same value which is used for filtered ephemerides. It makes no sense to use for example 15 degrees for GNSS and 5 degrees for filtered ephemerides, because ephemerides for satellites between 5 and 15 degrees would be downloaded, but those would not be used by GNSS. The other way around it makes more sense, you could download ephemerides for satellites above 15 degrees, but leave the GNSS threshold at 5 degrees. In this case GNSS would still be able to use satellites between 5 and 15 degrees by downloading the missing ephemerides from the broadcast, if satellites above 15 degrees are not enough for a fix.

All in all, my opinion is that this function should be removed. It would be good to disable scheduled downloads when A-GPS is used, but maybe there should be a separate PR for that instead. And I'm not sure if it's necessary to change the GNSS elevation threshold at all. It can be different from NRF_CLOUD_AGPS_ELEVATION_MASK and in most cases there's no need to change the default value of 5 deg.

Reply:
It would not make sense to filter to 15 degrees but leave the GNSS threshold to 5 -- the whole point of enabling A-GPS is to eliminate the need to wait for the GNSS unit to download ephemerides from the broadcast.  I think we discussed this elsewhere (in a call?) and it was decided that the two values should match.

Yes, I agree it doesn't belong here really (see my comment earlier: https://github.com/nrfconnect/sdk-nrf/pull/6388#discussion_r769011704).  But there is no other place I could find in the sdk-nrf repo that has any appropriate place to put such a general GNSS function.  I think there should be one that helps set the use case based on whether the NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE is supported or not, or at least, to determine if the modem firmware present supports a given feature.  Is there such a facility?

Reply:
Usually, there should be no need to change the elevation threshold from the default value. I think we at least discussed that the default for filtered ephemeris download should be the same which is used by the modem (5 deg). Having it configurable in the library is fine, but I don't think there's need for a function which configures the GNSS elevation threshold. Changing the elevation threshold with the GNSS API is a single function call with only one parameter.

There's no helper for setting the use case bitmask. There could be one for the reason you mentioned, but we don't have a good place for it, I think. However, I don't see this as a big problem, because if the modem FW doesn't support `NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE` and setting the use case bitmask fails, in most cases the only other set bit is `NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START` and this is already enabled by default. So there's no need to call `nrf_modem_gnss_use_case_set(NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START)` in the failure case. Actually, the `NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START` bit doesn't (currently) affect GNSS behavior at all, it just has been present in the API from the start and although we considered removing it, it could have caused some problems.

Reply:
I've tried to accommodate most of your suggestions.  Please review again and approve if possible.